kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ImplicitCastExpression.java

```diff
@@ -259,4 +262,15 @@ public double getDouble(int rowId) {
       return inputVector.getFloat(rowId);
     }
   }
+
+  private static class StringUpConverter extends UpConverter {
+    StringUpConverter(DataType targetType, ColumnVector inputVector) {
+      super(targetType, inputVector);
+    }
+
+    @Override
+    public String getString(int rowId) {
+      return inputVector.getString(rowId);
+    }
```

Comment:
why is this class needed? Seems like the same as the existing column vector.

Reply:
it is needed for converting collated StringTypes to non-collated and vice versa
i added it because all other casts have same classes